qcodes/instrument/base.py

```diff
@@ -401,6 +410,94 @@ def validate_status(self, verbose: bool = False) -> None:
                     print(f'validate_status: param {k}: {value}')
                 p.validate(value)
 
+    def _add_params_from_decorated_methods(self):
+
+        def special_kwargs_to_meth(self, kwarg_value):
+            """
+            Returns the attribute that has the name `kwarg_value.attr_name`
+            of this instance if the kwarg_value is of type `InstanceAttr`.
+            """
+            if isinstance(kwarg_value, InstanceAttr):
+                return getattr(self, kwarg_value.attr_name)
```

Comment:
perhaps it's useful to handle the case of the specified attribute not existing with an informative error message to users?